File: frontend/src/types/question.ts
export type GeneratedBy = "agent" | "human_edited";

export interface Question {
    id: string;
    job_id: string;
    criterion_id: string;
    content: string;
    rationale: string;
    generated_by: GeneratedBy;
    is_sensitive_flagged: boolean;
    sensitive_flag_reason: string | null;
    is_approved: boolean;
}

export interface QuestionUpdatePayload {
    content?: string;
    is_approved?: boolean;
}

File: backend/app/schemas/question.py
import uuid

from pydantic import BaseModel, ConfigDict


class QuestionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    job_id: uuid.UUID
    criterion_id: uuid.UUID
    content: str
    rationale: str
    generated_by: str
    is_sensitive_flagged: bool
    sensitive_flag_reason: str | None
    is_approved: bool


class QuestionUpdate(BaseModel):
    """HR sửa nội dung câu hỏi và/hoặc duyệt. Sửa content -> tự set generated_by=human_edited."""

    content: str | None = None
    is_approved: bool | None = None